### PR TITLE
VPP agent changes:

### PR DESCRIPTION
**Pull request by shastrinator**
1. VPP dhcp driver and host-interface changes
2. Restartability for gbp-vpp-agent
3. More interface settings: state/mtu
4. Adopting vpplib in opflexagent path
5. vpplib connection fixes

### CODE REVIEW

File: opflexagent/utils/ep_managers/endpoint_file_manager.py [hunk 1]
```diff
@@ -226,6 +240,14 @@ def _setup_ep_directory(self):
                     filename = f[:-len(FILE_EXTENSION) - 1]
                     if '_' in f:
                         self._registered_endpoints.add(f.split('_')[0])
+                        fp = file(os.path.join(directory, f))
+                        ep_opts = json.load(fp)
```
Comment (amittbose) [inline opflexagent/utils/ep_managers/endpoint_file_manager.py]: you need to gracefully handle json parsing errors here?

File: opflexagent/utils/ep_managers/endpoint_file_manager.py [hunk 2]
```diff
@@ -191,19 +193,31 @@ def declare_endpoint(self, port, mapping):
...
+            del self.vif_int_dict[port_id]
 
     def get_registered_endpoints(self):
         return self._registered_endpoints
 
     def get_stale_endpoints(self):
         return self._stale_endpoints
 
+    def get_access_int_for_vif(self, vif):
+        access_int = None
```
[the remark this answers is not in the document]
Comment (shastrinator) [reply opflexagent/utils/ep_managers/endpoint_file_manager.py]: ok. agree.

File: opflexagent/utils/ep_managers/endpoint_file_manager.py (hunk 1 above)
Comment (shastrinator) [reply opflexagent/utils/ep_managers/endpoint_file_manager.py]: ok

File: opflexagent/utils/ep_managers/endpoint_file_manager.py [hunk 3]
```diff
@@ -191,19 +193,31 @@ def declare_endpoint(self, port, mapping):
...
                                   mac_exceptions=macs)
             self._registered_endpoints.add(port.vif_id)
+            self.vif_int_dict.update({port.vif_id: port.port_name})
 
     def undeclare_endpoint(self, port_id):
         LOG.info("Endpoint undeclare requested for port %s", port_id)
         self._mapping_cleanup(port_id)
         self._registered_endpoints.discard(port_id)
         self._stale_endpoints.discard(port_id)
+        if port_id in self.vif_int_dict:
```
[the remark this answers is not in the document]
Comment (shastrinator) [reply opflexagent/utils/ep_managers/endpoint_file_manager.py]: ok..

**Comment by shastrinator** [comment]
I had the same thought too. It was Mandeep's decision that we make a separate repo along with UT later for vpplib later( I have an issue for this already). For the sake of making progress on integration, we want to have vpplib in opflexagent path for now.

**Pull request merged**